frontend/src/components/modal/column/ColumnOptionModal.vue

```diff
@@ -0,0 +1,216 @@
+<template>
+  <v-row>
+    <v-dialog v-model="dialog" persistent class="dialog">
+      <v-card class="card">
+        <v-row class="left-pad">
+          <v-col cols="12" sm="8">
+            <h2>Customize Columns</h2>
+            <small>Drag and Drop Columns to Change Order</small>
+          </v-col>
+          <v-col cols="12" sm="4">
+            <v-btn @click="addColumn"> Add Column </v-btn>
+          </v-col>
+        </v-row>
+        <v-row>
+          <v-col cols="12" sm="2">
+            <v-row v-for="i in cols.length" :key="i" class="number-rows">
+              <v-col cols="12" sm="3">
+                <h2>{{ i }}</h2>
+              </v-col>
+            </v-row>
+          </v-col>
+          <v-col cols="12" sm="10">
+            <v-card-text>
+              <draggable
+                :list="cols"
+                :animation="200"
+                ghost-class="ghost-card"
+                group="column.id"
+                id="column"
+              >
+                <ColumnCard
+                  v-for="column in cols"
+                  :key="column.id"
+                  :column="column"
+                  @deleteColumn="deleteColumn"
+                  @updateColumn="updateColumn"
+                  :tryError="invalidColumns"
+                />
+              </draggable>
+            </v-card-text>
+          </v-col>
+        </v-row>
+        <h4 class="error-message left-pad" v-if="unableToDeleteCol">
+          ** Unable to Delete Non-Empty Column **
+        </h4>
+        <h4 class="error-message left-pad" v-if="maxColumnsReached">
+          ** A Maximum of {{ MAX_COLS }} Columns Are Supported **
+        </h4>
+        <h4 class="error-message left-pad" v-if="minColumnsReached">
+          ** A Minimum of {{ MIN_COLS }} Column is Required **
+        </h4>
+        <h4 class="error-message left-pad" v-if="invalidColumns">
+          ** Ensure Each Column Has a Non-Empty Title **
+        </h4>
+        <v-card-actions>
+          <v-spacer></v-spacer>
+          <v-btn
+            color="blue-darken-1"
+            variant="text"
+            @click="this.closeClicked()"
+          >
+            Close
+          </v-btn>
+          <v-btn
+            color="blue-darken-1"
+            variant="text"
+            @click="this.saveClicked()"
+          >
+            Save
+          </v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-dialog>
+  </v-row>
+</template>
+
+<script>
+import ColumnCard from './ColumnCard.vue'
+import { VueDraggableNext } from 'vue-draggable-next'
+import { ref } from 'vue'
+
+const MAX_COLS = 8
+const MIN_COLS = 1
+const cols = ref([])
+const nextColId = ref(0) // set to max of existing cols + 1
```

Comment:
Is this corresponding to column id in the backend? Or column num.

In the case of column num, I'd say rename the item to reflect that.

Reply:
Just a temp value until backend integration is complete - will note in todos